packages/components/src/FilterBar/FilterBar.spec.tsx

```diff
@@ -752,4 +754,150 @@ describe("<FilterBar />", () => {
+            setItems(fetchedItems)
+          }
+        })
+      }, [JSON.stringify(activeFilterVals)])
+
+      return (
+        <FilterBar.MultiSelect id={id} items={items}>
+          {() => (
+            <FilterMultiSelect.ListBox>
+              {({ allItems }) => (
+                <FilterMultiSelect.ListBoxSection
+                  items={allItems}
+                  sectionName="All Items"
+                >
+                  {item => (
+                    <FilterMultiSelect.Option key={item.key} item={item} />
+                  )}
+                </FilterMultiSelect.ListBoxSection>
+              )}
+            </FilterMultiSelect.ListBox>
+          )}
+        </FilterBar.MultiSelect>
+      )
+    }
+
+    const fetchCityOptions = jest.fn((filterValues: Partial<AsyncValues>) => {
+      const isSupermanInFilterValue = Object.values(filterValues).some(
+        vals => vals.indexOf("3") > -1
+      )
+      const isBatmanInFilterValue = Object.values(filterValues).some(
+        vals => vals.indexOf("4") > -1
+      )
+
+      if (isBatmanInFilterValue && !isSupermanInFilterValue) {
+        return Promise.resolve([{ value: "1", label: "Gotham" }])
+      }
+
+      return Promise.resolve([
+        { value: "1", label: "Gotham" },
+        { value: "2", label: "Metropolis" },
+      ])
+    })
+
+    const fetchHeroOptions = jest.fn((filterValues: Partial<AsyncValues>) => {
+      const isGothamInFilterValue = Object.values(filterValues).some(
+        vals => vals.indexOf("1") > -1
+      )
+      const isMetroInFilterValue = Object.values(filterValues).some(
+        vals => vals.indexOf("2") > -1
+      )
+
+      if (isGothamInFilterValue && !isMetroInFilterValue) {
+        return Promise.resolve([{ value: "4", label: "Batman" }])
+      }
+
+      return Promise.resolve([
+        { value: "3", label: "Superman" },
+        { value: "4", label: "Batman" },
+      ])
+    })
+
+    const config = [
+      {
+        id: "city",
+        name: "City",
+        Component: (
+          <MockFilterAsyncComponent id="city" fetcher={fetchCityOptions} />
+        ),
+      },
+      {
+        id: "hero",
+        name: "Hero",
+        Component: (
+          <MockFilterAsyncComponent id="Hero" fetcher={fetchHeroOptions} />
+        ),
+      },
+    ] satisfies Filters<AsyncValues>
+
+    it("can re-fetch options with all active filter values pulled off of the FilterBarContext", async () => {
+      const { getByRole, queryByRole } = render(
+        <FilterBarWrapper<AsyncValues> filters={config} defaultValues={{}} />
+      )
+
+      // open city filter
+      await user.click(getByRole("button", { name: "City" }))
+
+      // assert both cities are shown
+      await waitFor(() => {
+        expect(queryByRole("option", { name: "Gotham" })).toBeVisible()
+        expect(queryByRole("option", { name: "Metropolis" })).toBeVisible()
+      })
+
+      // select filter option
```

Comment:
These comments are nice, but actually unnecessary since the code is self-documenting (except maybe the close filter one, that one isn't quite so clear reading just the code) :)